libr/core/cmd_search.c

```diff
@@ -1862,27 +1863,65 @@ static void rop_kuery(void *data, const char *input) {
 
 	switch (*input) {
 	case 'q':
-		sdb_list = sdb_foreach_list (db_rop);
-		ls_foreach (sdb_list, sdb_iter, kv) {
-			r_cons_printf ("%s ", kv->key);
+		ls_foreach (db_rop->ns, it, ns) {
+			sdb_list = sdb_foreach_list (ns->sdb);
+			ls_foreach (sdb_list, sdb_iter, kv) {
+				r_cons_printf ("%s ", kv->key);
+			}
 		}
-		r_cons_newline ();
 		break;
 	case 'j':
 		r_cons_print ("{\"gadgets\":[");
-		sdb_list = sdb_foreach_list (db_rop);
-		ls_foreach (sdb_list, sdb_iter, kv) {
-			if (json_first) {
-				json_first = false;
-			} else {
-				r_cons_print (",");
+			ls_foreach (db_rop->ns, it, ns) {
+			sdb_list = sdb_foreach_list (ns->sdb);
+			ls_foreach (sdb_list, sdb_iter, kv) {
+				if (json_first) {
+					json_first = false;
+				} else {
+					r_cons_print (",");
+				}
+				r_cons_printf ("{\"address\":%s,\"size\":%s}", kv->key, kv->value);
 			}
-			r_cons_printf ("{\"address\":%s,\"size\":%s}", kv->key, kv->value);
 		}
 		r_cons_printf ("]}\n");
 		break;
+	case ' ':
+		if (!strcmp (input + 1, "nop")) {
+			out = sdb_querys (core->sdb, NULL, 0, "rop/nop/*");
+			if (out) {
+				r_cons_println (out);
+			}
+			free (out);
+		} else if (!strcmp (input + 1, "mov")) {
+			out = sdb_querys (core->sdb, NULL, 0, "rop/mov/*");
+			if (out) {
+				r_cons_println (out);
+			}
+			free (out);
+		} else if (!strcmp (input + 1, "const")) {
+			out = sdb_querys (core->sdb, NULL, 0, "rop/const/*");
+			if (out) {
+				r_cons_println (out);
+			}
+			free (out);
+		} else if (!strcmp (input + 1, "arithm")) {
+			out = sdb_querys (core->sdb, NULL, 0, "rop/arithm/*");
+			if (out) {
+				r_cons_println (out);
+			}
+			free (out);
+		} else if (!strcmp (input + 1, "arithm_ct")) {
+			out = sdb_querys (core->sdb, NULL, 0, "rop/arithm_ct/*");
+			if (out) {
+				r_cons_println (out);
+			}
+			free (out);
```

Comment:
as long as you are checking for 'out' already you can place the free inside the conditional block, same for the other cases